core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -686,18 +772,33 @@ private ApplicationManager createSourceAndDeployApp(String appName, File file, S
     if (schema != null) {
       schemaString = schema.toString();
     }
-    ETLStage source = new ETLStage(
-      "source", new ETLPlugin("File", BatchSource.PLUGIN_TYPE,
-                              ImmutableMap.<String, String>builder()
-                                .put(Constants.Reference.REFERENCE_NAME, "TestFile")
-                                .put(Properties.File.FILESYSTEM, "Text")
-                                .put(Properties.File.PATH, file.getAbsolutePath())
-                                .put(Properties.File.FORMAT, format)
-                                .put(Properties.File.IGNORE_NON_EXISTING_FOLDERS, "false")
-                                .put("pathField", "file")
-                                .put(Properties.File.SCHEMA, schemaString)
-                                .build(),
-                              null));
+    ETLStage source;
+    if (schema != null) {
```

Comment:
this is unnecessary repeat code. The only thing you need to do if schema is not null is put the schema property in the ImmutableMap builder